### DISCUSSION TITLE
What would be the goals of OpenPharmacophore?

### DISCUSSION THREAD

**Discussion by dprada** [uibcdf/OpenPharmacophore#1] [Ideas]
Let's discuss here what would be the functionalities of OpenPharmacophore.

**Comment by dprada** [comment]
First of all, the library should be an open-source free tool to build pharmacophore models to be used for virtual screening. The library should be extremely modular to be used, in its simple version, as the platform to implement and try easily new approaches and modifications.

**Reply by dprada** [reply]
One of the main features of this library I forgot to add is 'modeling from molecular dynamics'.

**Comment by dprada** [comment]
The library could, in the future, be used to make *de novo* design instead of only virtual screening?

**Reply by Daniel-Ibarrola** [reply]
Sure, I think it could be a good idea to implement in the future.

**Comment by dprada** [comment]
Would the library include a set of methods to extract pharmacophoric points from chemical probes molecular dynamics?

**Reply by Daniel-Ibarrola** [reply]
It could include different methods and algorithms to generate pharmacophores. That way different approaches could be used in case one does not work for a particular problem.

**Reply by dprada** [reply]
This point can be something that makes openpharmacophore different from any other software. Usually, what we can find out there is a predefined workflow. Every software has its own approach. But we could implement different options, again, very modular. This way anyone can define the best combination. As you said, not every system should be tackled with the same recipe.

**Comment by dprada** [comment]
The library has to include the possibility of distributing the tasks to threads in a machine or the resources of a local HPC cluster. See LigandScout Remote:

[Kainrad, Thomas, Sascha Hunold, Thomas Seidel, and Thierry Langer. “LigandScout Remote: A New User-Friendly Interface for HPC and Cloud Resources.” *Journal of Chemical Information and Modeling* 59, no. 1 (2018): 31–37.](https://doi.org/10.1021/acs.jcim.8b00716)

**Comment by Daniel-Ibarrola** [comment]
Perhaps, it should include capabilities to work with different types of files. For example, some papers mention the use of protein data bank files, or that they use files that then are converted to other file types so they can be used with other software.

**Reply by dprada** [reply]
That's also a good point. It should be very agnostic. To do this, we will work with MolSysMT. An in-house developed library that should be very useful supporting OpenPharmacophore. Let's talk about MolSysMT next Monday (May 3rd).
